### PR TITLE
Template of working with databases PR

### PR DESCRIPTION
**Pull request by Piachsecki**

### CODE REVIEW

File: src/main/java/org/example/domain/customer/Customer.java [hunk 1]
```diff
@@ -1,21 +1,339 @@
...
+    }
+
+    public Customer withEntrepreneurshipForm(Entrepreneurship entrepreneurshipForm) {
+        this.entrepreneurshipForm = entrepreneurshipForm;
+        return this;
+
+    }
+
+    //this will stay so, to discuss the importance of the CountMonthlyRevenueService
+    public Money countMonthlyRevenue(YearMonth yearMonth) {
```
Comment (biellukasz) [inline src/main/java/org/example/domain/customer/Customer.java:74]: i to jest spoko

File: src/main/java/org/example/domain/customer/Customer.java [hunk 2]
```diff
@@ -1,21 +1,339 @@
...
+    }
+
+    public Money countIncomeTax(YearMonth yearMonth) {
+        Money monthlyRevenue = countMonthlyRevenue(yearMonth);
+        Money monthlyCosts = countMonthlyCosts(yearMonth);
+
+        BigDecimal incomeBeforeTaxIncludingSocialContributions = countIncomeBeforeTaxIncludingSocialContributions(monthlyRevenue, monthlyCosts);
+        TaxPaymentForm customerTaxPaymentForm = getCustomerTaxPaymentForm(this);
+
+        if (customerTaxPaymentForm instanceof FlatTax) {
```
Comment (biellukasz) [inline src/main/java/org/example/domain/customer/Customer.java:142]: switch case bedzie czytelniesze?

File: src/main/java/org/example/domain/customer/Customer.java [hunk 3]
```diff
@@ -1,21 +1,339 @@
...
+        if (monthlyRevenue.compareTo(FIRST_CONTRIBUTION_RANGE) < 0) {
+            return FIRST_RANGE;
+        } else if (monthlyRevenue.compareTo(FIRST_CONTRIBUTION_RANGE) >= 0 &&
+                monthlyRevenue.compareTo(SECOND_CONTRIBUTION_RANGE) < 0) {
+            return SECOND_RANGE;
+        } else {
+            return THIRD_RANGE;
+        }
+
+    }
```
Comment (biellukasz) [inline src/main/java/org/example/domain/customer/Customer.java:324]: zrobila sie troche duza klasa ale chyba taka ma byc

File: src/main/java/org/example/Data.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.example;
+
+import java.math.BigDecimal;
+
+public class Data {
```
Comment (biellukasz) [inline src/main/java/org/example/Data.java:5]: Dane czego?

File: src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import java.util.Optional;
+import java.util.UUID;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@Slf4j
+public class AccountingOfficeDatabaseTest {
+    private CustomerRepository customerRepository;
+    private InvoiceRepository invoiceRepository;
+    private InsertInvoiceService insertInvoiceService;
+    private
```
Comment (biellukasz) [inline src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java:35]: ?

File: src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    @BeforeEach
+    void initClasses(){
```
Comment (biellukasz) [inline src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java:38]: do jakiej bazy sie tutaj laczysz? masz cos lokalnie postawione?

File: src/test/java/org/example/domain/customer/CustomerTest.java [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+import java.math.BigDecimal;
+import java.time.YearMonth;
+
+import static org.example.Data.*;
+import static org.example.domain.customer.TaxPayments.FlatTax.FLAT_TAX_RATE;
+import static org.example.domain.customer.TaxPayments.GeneralTax.FIRST_SCALE;
+import static org.example.domain.customer.TaxPayments.GeneralTax.SECOND_SCALE;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomerTest {
```
Comment (biellukasz) [inline src/test/java/org/example/domain/customer/CustomerTest.java:25]: git

**Review by biellukasz — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java (hunk 6 above)
Comment (Piachsecki) [reply src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java:38]: I had a local database set up -> In the next part of the project going to use migration and docker to store my database in there.
Resolved

File: src/main/java/org/example/adapter/out/database/entity/AddressDatabaseEntity.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Getter
+@Setter
+@EqualsAndHashCode(of = "addressId")
+@Entity
+@Table(name  = "address")
+public class AddressDatabaseEntity {
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/adapter/out/database/entity/AddressDatabaseEntity.java:15]: These classes have to be public for the mapping purposes. -> mapping customerDatabaseEntity into customer and so on...
resolved.

File: src/main/java/org/example/adapter/out/database/repository/CustomerDatabaseStorage.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.UUID;
+
+import static org.example.adapter.out.database.repository.EntityToDomainClassMapper.createCustomerFromCustomerDatabaseEntity;
+
+@Slf4j
+public class CustomerDatabaseStorage implements CustomerRepository {
+    @Override
+    public Customer addCustomer(Customer customer) {
+        try (Session session = DatabaseHibernateConfig.getSession()) {
+            if (Objects.isNull(session)) {
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/adapter/out/database/repository/CustomerDatabaseStorage.java:24]: It is an additional check for the program to start.
resolved.

File: src/main/java/org/example/application/FindUserByNIPService.java [hunk 10]
```diff
@@ -1,12 +1,15 @@
...
 import org.example.port.in.customer.FindCustomerUseCase;
 import org.example.port.out.CustomerRepository;
 
 import java.util.Optional;
 
+@AllArgsConstructor
+
 public class FindUserByNIPService implements FindCustomerUseCase {
 
     private CustomerRepository customerRepository;
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/application/FindUserByNIPService.java:15]: Changed the whole structure of the service.
Resolved.

File: src/main/java/org/example/application/InsertCostInvoiceService.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+import org.example.port.out.InvoiceRepository;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+@AllArgsConstructor
+
+public class InsertCostInvoiceService implements InsertCostInvoiceUseCase {
+    private InvoiceRepository invoiceRepository;
+    @Override
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/application/InsertCostInvoiceService.java:20]: ?

File: src/main/java/org/example/domain/customer/TaxPayments/GeneralTax.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
+package org.example.domain.customer.TaxPayments;
+
+import java.math.BigDecimal;
+
+public class GeneralTax extends TaxPaymentForm{
+    public static final String FIRST_SCALE = "0.12";
+    public static final String SECOND_SCALE = "0.32";
+    public static final BigDecimal SCALE_SALARY = BigDecimal.valueOf(5000);
+    public GeneralTax() {
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/domain/customer/TaxPayments/GeneralTax.java:9]: resolved.

File: src/main/java/org/example/domain/customer/TaxPayments/HEALTH_INSURANCE_RANGE_FOR_LUMP_SUM.java [hunk 13]
```diff
@@ -0,0 +1,7 @@
+package org.example.domain.customer.TaxPayments;
+
+public enum HEALTH_INSURANCE_RANGE_FOR_LUMP_SUM {
```
[the remark this answers is not in the document]
Comment (Piachsecki) [reply src/main/java/org/example/domain/customer/TaxPayments/HEALTH_INSURANCE_RANGE_FOR_LUMP_SUM.java:3]: changed.
resolved.

File: src/main/java/org/example/domain/customer/Customer.java (hunk 2 above)
Comment (Piachsecki) [reply src/main/java/org/example/domain/customer/Customer.java:142]: We discussed on our previous meeting.
Resolved.

File: src/main/java/org/example/Data.java (hunk 4 above)
Comment (Piachsecki) [reply src/main/java/org/example/Data.java:5]: Static constants used in the program.
resolved.

File: src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java (hunk 5 above)
Comment (Piachsecki) [reply src/test/java/org/example/adapter/out/database/AccountingOfficeDatabaseTest.java:35]: I dont know how to check the actual connection and queries asked to the database itself, thats why I imitate the connection and methods in test class, where by order of the methods I am testing single functionalities. Everything is checked now.
Resolved.

**Pull request closed without merging**